Library/Homebrew/test/test_ENV.rb

```diff
@@ -141,4 +141,32 @@ def test_initializes_deps
     assert_equal [], @env.deps
     assert_equal [], @env.keg_only_deps
   end
+
+  def test_unsupported_cxx11
+    ["gcc", "gcc-4.7"].each do |compiler|
+      @env["HOMEBREW_CC"] = compiler
+      assert_raises do
+        @env.cxx11
+      end
+      assert_nil @env["HOMEBREW_CCCFG"]
+    end
+  end
+
+  def test_supported_cxx11_gcc_5
+    @env["HOMEBREW_CC"] = "gcc-5"
+    @env.cxx11
+    assert_equal @env["HOMEBREW_CCCFG"], "x"
```

Comment:
In the same vein like the other comment, let's make as few assumptions as possible. Also, the expected value is supposed to be passed as the first argument. Hence, please make this:

```ruby
assert_match "x", @env["HOMEBREW_CCCFG"]
```

(Also applies to the following two instances of basically the same check.)